prql-compiler/src/sql/srq/anchor.rs

```diff
@@ -57,34 +60,16 @@ pub(super) fn extract_atomic(
     atomic
 }
 
-/// Converts a pipeline into atomic pipelines. Meant for debugging purposes.
-pub(super) fn extract_atomics_naive(
-    pipeline: Vec<SqlTransform>,
-    ctx: &mut AnchorContext,
-) -> Vec<Vec<SqlTransform>> {
-    let mut atomics = Vec::new();
-
-    atomics.push(extract_atomic(pipeline, ctx));
-    while let Some(decl_key) = ctx.table_decls.keys().max_by_key(|t| t.get()).cloned() {
-        let decl = ctx.table_decls.remove(&decl_key).unwrap();
-        if let Some(relation) = decl.relation {
-            if let SqlRelationKind::PreprocessedPipeline(p) = relation.kind {
-                atomics.push(extract_atomic(p, ctx));
-            }
-        }
-    }
-
-    atomics.reverse();
-    atomics
-}
-
 /// Splits pipeline into two parts, such that the second part contains
 /// maximum number of transforms while "fitting" into a SELECT query.
```

Comment:
```suggestion
/// maximum number of transforms while "fitting" into a SELECT query,
/// and the first part contains the remainder. The input pipeline is consumed.
```

(nothing groundbreaking but for my own notes, feel free to discard)